src/program/latex/mod.rs

```diff
@@ -0,0 +1,1600 @@
+    /// a BTreeMap of wires with the name of the wire as the key
+    circuit: BTreeMap<u64, Box<Wire>>,
+}
+
+impl Default for Diagram {
+    fn default() -> Self {
+        Self {
+            settings: Settings::default(),
+            column: 0,
+            relationships: HashMap::new(),
+            circuit: BTreeMap::new(),
+        }
+    }
+}
+
+impl Diagram {
+    /// Compares qubits from a single instruction associated with a column on
+    /// the circuit to all of the qubits used in the quil program. If a qubit
+    /// from the quil program is not found in the qubits in the single
+    /// instruction line, then an empty slot is added to that column on the
+    /// qubit wire of the circuit.
+    ///
+    /// # Arguments
+    /// `&mut self` - exposes the Circuit
+    /// `qubits` - qubits used in the quil program
+    /// `instruction` - exposes qubits in a single instruction
+    fn set_qw(&mut self, qubits: &HashSet<Qubit>, instruction: &Instruction) {
+        for program_qubit in qubits {
+            let mut found = false;
+            match &instruction {
+                instruction::Instruction::Gate(gate) => {
+                    for gate_qubit in &gate.qubits {
+                        if program_qubit == gate_qubit {
+                            found = true;
+                        }
+                    }
+
+                    if !found {
+                        match program_qubit {
+                            instruction::Qubit::Fixed(q) => {
+                                self.circuit
+                                    .get_mut(&q)
+                                    .and_then(|wire| wire.empty.insert(self.column, Command::Qw));
+                            }
+                            _ => (),
+                        }
+                    }
+                }
+                _ => (),
+            }
+        }
+    }
+
+    /// Returns a reformatted gate name based on the modifiers used in a single
+    /// instruction line of a quil program or the original name. Gates with
+    /// CONTROLLED modifiers are reformatted such that each CONTROLLED modifier
+    /// prepends a `C` to the beginning of the gate name. For other modifiers
+    /// such as DAGGER, no special reformatting is required.
+    ///
+    /// For example, for an instruction line `CONTROLLED CONTROLLED Y 0 1 2` the
+    /// gate name is reformatted to `CCY` where each C is mapped to an
+    /// associated qubit with the last qubit to the original gate name. For an
+    /// instruction line `DAGGER DAGGER Y 0`, the gate name remains `Y`,
+    /// instead each of the modifiers are added to the wire at the current
+    /// column where it is to be applied using the Command::Super variant.
+    ///
+    /// # Arguments
+    /// `&mut self` - exposes the current column of the Circuit
+    /// `gate` - exposes the modifiers associated with the instruction gate
+    /// `wire` - exposes the wire to be pushed to in Circuit
```

Comment:
Again, don't think this is necessary

Reply:
Removed the middle paragraph of the docs [Use write! in Diagram fmt, reduce/reformat docs.](https://github.com/rigetti/quil-rs/pull/145/commits/6a2fd9e508c61782156ac7e9e3829320153c0df8)